### PR TITLE
Fail gracefully if main_menu_script has bad value

### PR DESCRIPTION
**Pull request by Wuzzy2**
Partial fix of #9003. If `main_menu_script` has an invalid value, Minetest will no longer crash, but instead fall back to the default main menu script and prints a warning in log.

## To do

This PR is ready for review.

## How to test

1. Set `main_menu_script` to an invalid value
2. Start Minetest
3. Check log

### CODE REVIEW

File: builtin/init.lua [hunk 1]
```diff
@@ -39,9 +39,20 @@ if INIT == "game" then
...
-		dofile(mm_script)
-	else
+		local testfile = io.open(mm_script, "r")
+		if testfile then
+			dofile(mm_script)
+			testfile:close()
+			custom_loaded = true
+			core.log("info", "Loaded custom main menu script: "..mm_script)
+		else
+			core.log("warning", "Failed to load custom main menu script: "..mm_script)
```
Comment (rubenwardy) [inline builtin/init.lua:51]: should be an error

**Comment by Wuzzy2** [comment]
Warning changed to error.

**Review by SmallJoker — APPROVED** [review]
LGTM

File: builtin/init.lua [hunk 2]
```diff
@@ -39,9 +39,20 @@ if INIT == "game" then
 	assert(not core.get_http_api)
 elseif INIT == "mainmenu" then
 	local mm_script = core.settings:get("main_menu_script")
+	local custom_loaded = false
 	if mm_script and mm_script ~= "" then
-		dofile(mm_script)
-	else
+		local testfile = io.open(mm_script, "r")
+		if testfile then
+			dofile(mm_script)
```
Comment (rubenwardy) [inline builtin/init.lua:46]: I'd put `testfile:close()` before the dofile

**Comment by Wuzzy2** [comment]
@rubenwardy: Done.

**Pull request merged**